lib/src/components/NxToggle/__tests__/NxToggle.test.tsx

```diff
@@ -21,46 +17,38 @@ describe('NxToggle', function() {
     children: 'Enables whales'
   };
 
-  const rtlRender = rtlUtils.rtlRender<Props>(NxToggle, simpleProps);
-
-  it('renders a <label> with an <input>', function() {
-    rtlRender();
-
-    const checkbox = screen.getByRole<HTMLInputElement>('switch', { name: 'Enables whales' });
+  const quickRender = rtlRender<Props>(NxToggle, simpleProps);
+  const renderEl = rtlRenderElement(NxToggle, simpleProps);
 
-    expect(checkbox.tagName).toEqual('INPUT');
-
-    const label = checkbox.labels?.[0];
-
-    expect(label?.textContent).toEqual('Enables whales');
+  it('renders a swith with the correct label', function() {
+    const checkbox = quickRender().getByRole('switch', { name: 'Enables whales' });
+    expect(checkbox).toBeInTheDocument();
   });
 
-  it('renders an input element with role switch with the correct attributes and classname', function() {
-    rtlRender();
-
-    const checkbox = screen.getByRole('switch');
-
-    expect(checkbox).toHaveAttribute('type', 'checkbox');
+  it('renders an input element with role=switch with the correct attributes', function() {
+    const checkbox = quickRender().getByRole('switch');
+    expect(checkbox).toBeInTheDocument();
     expect(checkbox).toHaveAttribute('id', 'toggle-id');
-    expect(checkbox).toHaveClass('nx-toggle__input');
   });
 
-  it('adds classes specified with the className prop', function() {
-    const { container } = rtlRender({ className: 'foo' });
+  it('merges any passed in className', function() {
+    const componentWithAddedClass = renderEl({ className: 'foo' });
+    const component = renderEl()!;
 
-    const root = container.children[0];
+    expect(componentWithAddedClass).toHaveClass('foo');
 
-    expect(root).toHaveClass('foo');
-    expect(root).toHaveClass('nx-toggle');
+    for (const cls of Array.from(component.classList)) {
+      expect(componentWithAddedClass).toHaveClass(cls);
+    }
   });
 
   it('calls its onChange prop when the label is clicked', async function() {
     const user = userEvent.setup();
 
     const onChange = jest.fn();
 
-    const { container } = rtlRender({ onChange });
-    const label = container.querySelector('label') as HTMLLabelElement;
+    const { container } = quickRender({ onChange });
+    const label = container.querySelector('label')!;
```

Comment:
I've wondered about this myself.  `<label>`s have a funny place in ARIA - they have no default role, may not be given a role, and may not be named.  They basically only exist to provide names to other elements, as far as aria is concerned.  At the same time, the fact that clicking a `<label>` will activate its associated form control is built into HTML.  These two facts together have caused me to conclude that we don't really need to test clicking on `<label>`s separately from clicking on their associated form fields.  So for this test I'd query the `switch` role and click that.  And not worry about the `<label>`